implement sponsors pages

OleksandrDragun opened the pull request:
### PR Checklist

* [x] Have you verified that the PR is pointing to the correct target branch? (`develop` for features/bugfixes, `master` for release/hotfix)?
* [x] Have you followed the guidelines in our [Dev Guide](../#readme)
* [x] Have you lint your code locally prior to submission (`npm run lint`)?
* [x] Have you run prettier (`npm run format`)?
* [ ] Have you checked that app can be built (`npm run build`)?
* [ ] Have you checked that the build output looks according to design (screen, mobile, desktop) in Figma (`npm run serve`)?

## Visuals

<!-- OPTIONAL
  Provide the visual proof (screenshot/gif/video) of your work
-->

AmsterGet commented on src/containers/SponsorsPage/IndividualPage/IndividualPage.tsx:
```diff
@@ -0,0 +1,31 @@
+      subtitle="By sponsoring ReportPortal, individuals can make a significant impact, helping to ensure the project's longevity and success."
+      cardsInfo={INDIVIDUAL_CARDS}
+    />
+    <SponsorshipTiers sponsorshipType="individual" />
```
I see that you use the `sponsorKind="individual"` for `WhySponsors` component and `sponsorshipType="individual"` for this `SponsorshipTiers`.
Does it make sense to use a single name for sponsorship type prop for these two components?

OleksandrDragun replied:
Done, thanks.

AmsterGet commented on src/containers/SponsorsPage/WhySponsors/icons/CommunityEngagement.tsx:
```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+
+export const CommunityEngagement = () => {
```
Can it be replaced by using `*.inline.svg`?
The same comment about other similar cases.

OleksandrDragun replied:
To be frank I just made WhyReportPortal component reusable with current styles and took implementation from there. 
I'm afraid of breaking something on that page.

OleksandrDragun replied:
Corrected, thanks.

ViktorSoroka07 commented on src/containers/SponsorsPage/HowToBecomeSponsor/HowToBecomeSponsor.tsx:
```diff
@@ -0,0 +1,94 @@
+const getStepsItemWith = createBemBlockBuilder(['steps-item']);
+
+export const HowToBecomeSponsor: FC<HowToBecomeSponsorProps> = ({ contactUsLink }) => {
+  const scrollToTiers = () => {
+    const anchorTarget = document.getElementById('sponsorship-tiers');
+
+    if (anchorTarget) {
+      anchorTarget.scrollIntoView({ behavior: 'smooth', block: 'start', inline: 'nearest' });
+    }
+  };
```
Let's create hook for that

hook should looks something like this

```
const useScrollIntoViewHandler = () => (id: string) => {
  const element = document.getElementById(id);

  if (anchorTarget) {
    element.scrollIntoView({ behavior: 'smooth', block: 'start', inline: 'nearest' });
  }
}
```

so it can be used like this


```
const scrollToViewHandler = useScrollToViewHandler('sponsorship-tiers');

...

onClick=(() => scrollToViewHandler('sponsorship-tiers'))
```

Please use it also on `FeaturesPage`

OleksandrDragun replied:
Done.

ViktorSoroka07 commented on src/containers/SponsorsPage/HowToBecomeSponsor/HowToBecomeSponsor.tsx:
```diff
@@ -0,0 +1,94 @@
+              ReportPortal{' '}
+              <Link
+                className={getStepsItemWith('__link')}
+                to="https://github.com/sponsors/reportportal"
```
Let's store it to variable and use in both places

OleksandrDragun replied:
Done.

ViktorSoroka07 commented on src/containers/SponsorsPage/SponsorshipTiers/SponsorshipTiers.tsx:
```diff
@@ -0,0 +1,28 @@
+      subtitle="Choose your level of sponsorship using GitHub and gain access to exclusive benefits "
+    />
+    <div className={getBlocksWith('__card-wrapper')}>
+      {SPONSORSHIP_INFO[sponsorshipType].map(info => (
+        <PricingCard key={info.key} {...info} />
```
```suggestion
      {SPONSORSHIP_INFO[sponsorshipType].map(({ key, ...info })=> (
        <PricingCard key={key} {...info} />
```

OleksandrDragun replied:
Done.

ViktorSoroka07 commented on src/containers/SponsorsPage/IndividualPage/IndividualPage.tsx:
```diff
@@ -0,0 +1,32 @@
+        title="Still have questions about sponsorship?"
+        linkTitle="Contact us"
+        link={CONTACT_US_LINK}
+        dataGtm="contact_us"
```
Let's move to constant `contact_us` so that if we need to change it we update only one place

OleksandrDragun replied:
Done.

ViktorSoroka07 commented on src/containers/SponsorsPage/SponsorshipTiers/SponsorshipTiers.tsx:
```diff
@@ -0,0 +1,28 @@
+import './SponsorshipTiers.scss';
+
+interface SponsorshipTiersProps {
+  sponsorshipType: 'business' | 'individual';
```
```suggestion
  sponsorshipType: keyof typeof SPONSORSHIP_INFO;
```

OleksandrDragun replied:
Done.

ViktorSoroka07 reviewed (commented):
Nice work!

ViktorSoroka07 commented:
Please fill PR Checklist. It is there for purpose

OleksandrDragun commented on src/components/OfferPageWrapper/OfferPageWrapper.tsx:
```diff
@@ -58,6 +58,7 @@ export const OfferPageWrapper: FC<OfferPageWrapperProps> = ({
   pricing,
 }) => {
   const { buttons, isDiscount, toggleDiscount } = usePricingHeroProps(page);
+  const { title, subtitle, description, offerType } = hero;
```
Of course I can keep it like before, but I tried to solve this TS problem
![image](https://github.com/reportportal/reportportal.github.io/assets/42969419/0310a2ba-a37e-4b32-a688-d74eb94a7393)
If you know the answer please let me know.

OleksandrDragun replied:
Done.

OleksandrDragun commented on src/components/Banner/Banner.tsx:
```diff
@@ -11,10 +11,18 @@ interface BannerProps {
   linkTitle?: string;
   subtitle?: string;
+  dataGtm?: string;
   children?: React.JSX.Element;
```
I absolutely agree. I'll correct it.

OleksandrDragun replied:
Done.

OleksandrDragun commented on src/containers/SponsorsPage/BusinessPage/constants.ts:
```diff
@@ -0,0 +1,33 @@
+import brandVisibilitySvg from './icons/brand-visibility.svg';
+import communityEngagementSvg from './icons/community-engagement.svg';
+import productImprovementSvg from './icons/product-improvement.svg';
+import talentAttractionSvg from './icons/talent-attraction.svg';
```
Done.

OleksandrDragun commented on src/utils/constants.ts:
```diff
@@ -83,6 +83,14 @@ export const SEO_DATA = {
   },
+  business: {
+    title: 'Business',
+    description: 'TBD',
```
Done.

OleksandrDragun commented on src/containers/SponsorsPage/WhySponsors/WhySponsors.tsx:
```diff
@@ -0,0 +1,19 @@
+import './WhySponsors.scss';
+
+interface WhySponsorsProps {
+  sponsorshipType: string;
```
Done.

OleksandrDragun commented on src/components/PricingCard/PricingCard.tsx:
```diff
@@ -0,0 +1,100 @@
+          {actionText}
+          {isAbsoluteURL(href) && <ArrowIcon />}
+        </button>
+      )}
```
Done.

OleksandrDragun commented on src/containers/SassPage/PricingCards/PricingCards.tsx:
```diff
@@ -12,15 +11,22 @@ interface PricingCardsProps {
-);
+export const PricingCards: FC<PricingCardsProps> = ({ pricing, isDiscount }) => {
+  const convertedPriceList = SAAS_OFFERS.map(offer => {
+    return {
```
Done.

OleksandrDragun commented on src/containers/SponsorsPage/BusinessPage/BusinessPage.scss:
```diff
@@ -0,0 +1,9 @@
+.business {
+  .linked-card-block__cards {
+    max-width: 1200px;
```
Done.

OleksandrDragun commented:
The command npm run build does not work locally 
![image](https://github.com/reportportal/reportportal.github.io/assets/42969419/0df98d43-fa07-4619-8c2d-46241bd4b0a6)
I've cleaned the cache couple of time and removed node_modules the result the same.
